Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp

```diff
@@ -173,15 +173,21 @@ int RGroupDecomposition::add(const ROMol &inmol) {
 
   std::vector<RGroupMatch> potentialMatches;
 
+  std::unique_ptr<ROMol> tMol;
+  std::unique_ptr<ROMol> coreCopy;
   for (auto &tmatche : tmatches) {
-    boost::scoped_ptr<ROMol> tMol;
-    {
-      const bool replaceDummies = false;
-      const bool labelByIndex = true;
-      const bool requireDummyMatch = false;
-      tMol.reset(replaceCore(mol, *rcore->core, tmatche, replaceDummies,
-                             labelByIndex, requireDummyMatch));
+    const bool replaceDummies = false;
+    const bool labelByIndex = true;
+    const bool requireDummyMatch = false;
+    coreCopy.reset(new ROMol(*rcore->core));
```

Comment:
I might consider making this a function for better documentation

```
/// Return the core with non label query atoms replaced with the atoms in the mathing molecule
std::unique_ptr<ROMol*> getCoreFromMatch() const { ...}
...
...

std::unique_ptr<ROMol*> coreCopy = rcore->getCoreFromMatch( tmatche );
tMol.reset(replaceCore(mol, *coreCopy, tmatche, replaceDummies,
                            labelByIndex, requireDummyMatch));
```

Reply:
@bp-kelley Done!